fetching/tarball-fetcher/src/index.ts

```diff
@@ -39,7 +42,14 @@ export function createTarballFetcher (
     offline?: boolean
   }
 ): TarballFetchers {
-  const download = createDownloader(fetchFromRegistry, {
+  const workerPool = new WorkerPool({
+    id: 'tarball',
+    maxWorkers: os.cpus().length - 1,
+    workerScriptPath: path.join(__dirname, 'worker/tarballWorker.js'),
+  })
+  // @ts-expect-error
+  global.finishWorkers = () => workerPool.finishAsync()
```

Comment:
This is a hack. For now it should be fine as I want to minimize the changes in this PR but in the future we'll have to find a better solution to call this function.